### ISSUE TITLE
Add input wrapper styling

### ISSUE THREAD

**Issue by cameronoelsen** [jupyterlab/jupyterlab#1472]
Currently in JupyterLab, we are unable to style the wrapper surrounding any input field in the interface when the input field is in focus. For example, I am trying to make it so that when a user is typing in the search field in the command palette, the interface will resemble this:

![screen shot 2017-01-10 at 11 42 32 am](https://cloud.githubusercontent.com/assets/6437976/21821851/e67b663a-d729-11e6-959d-d301e71e511b.png)

But currently
![screen shot 2017-01-10 at 11 44 12 am](https://cloud.githubusercontent.com/assets/6437976/21821915/2095cf54-d72a-11e6-9ad0-3903ef600c46.png)

@sccolbert suggested we add this to phosphor since this will be used all over the interface.

**Comment by sccolbert** [comment]
So, it could certainly make sense to add this to phosphor directly in the form of a `p-mod-focused` CSS class on the input wrapper. But is that enough? What if people want to style the whole command palette based on whether anything in the palette has focus? Then you have a `p-mod-focused` on the widget itself, and which point is it proper to add that to phosphor or leave it to the plugin? Discussion should be had on this.

**Comment by cameronoelsen** [comment]
It is certainly needed for all inputs throughout the interface. I am not sure though the extent in which we should extend the class. Pinging @blink1073 @ellisonbg for their input on this as well

**Comment by blink1073** [comment]
I think adding the CSS class on the input wrapper is enough.  The developer can add a focus listener  if they want do anything more advanced.

**Comment by blink1073** [comment]
For the ones added to dialogs, we already have a class for that: https://github.com/jupyterlab/jupyterlab/blob/24649b5d7e8501f79237124089fe9a84bcc8cac0/src/common/dialog.css#L126

**Comment by sccolbert** [comment]
Phosphor already adds a CSS class to the wrapper for the builtin widgets. Do you need anything from me on this?

**Comment by blink1073** [comment]
Per our conversation today, I am going to add a modifier to the input wrapper when it's input has focus. 

@sccolbert, that was the part I was missing in my previous comment.

**Comment by AbdealiJK** [comment]
I can see that there is a pull request that addresses this for the input tags in dialogs. But what about other input tags ? Like p-CommandPalette-input ? Does this same javascript for _evtBlur and _evtFocus need to be done in commandpallete plugin ?

And in any other plugin that uses input tags ?

**Comment by sccolbert** [comment]
@AbdealiJK The recent phosphor update fixes this for `CommandPalette`

**Comment by blink1073** [comment]
@AbdealiJK, if any of the other ones are missing the styling it is a bug.  We can fix the command palette styling after #1762 is merged.

**Comment by AbdealiJK** [comment]
Ah, nice. Got it.
